From 97d65a0046e64de94836fc44837dcef76961e616 Mon Sep 17 00:00:00 2001
From: Nana Guo <nana.n.guo@intel.com>
Date: Wed, 8 Apr 2015 17:37:35 -0400
Subject: [PATCH 12/12] remove HW JPEG decoder plugin

Changes:
remove JPEGMixImageDecoder plugin for HW JPEG

Change-Id: Idcd8e2aa99cb439d8831152b3cb5a4fa5094f2c2
Old-Change-ID: I426f3acec1d64363b24089f20a73ebd8b0270bd1
Ori-Change-ID: If97fd5bc648e645171f092ce970fc90a6684388a
Tracked-On: https://jira01.devtools.intel.com/browse/CTEAN-1950
Orig-Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-25838
Category: aosp improvement
Domain: Video.Media-jpeg
Origin: internal
Upstream-Candidate: no, proprietary
Signed-off-by: Nana Guo <nana.n.guo@intel.com>
Old-Reviewed-on: https://android.intel.com:443/374232
Signed-off-by: Xinyun Liu <xinyun.liu@intel.com>
---
 Android.mk                    |    2 +-
 include/core/SkImageDecoder.h |    1 -
 src/images/SkForceLinking.cpp |    1 -
 3 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/Android.mk b/Android.mk
index 7a097fb..149c1bc 100644
--- a/Android.mk
+++ b/Android.mk
@@ -535,7 +535,7 @@ LOCAL_STATIC_LIBRARIES := \
 ifeq ($(ADD_INTEL_SKIMAGEDECODER), true)
 LOCAL_STATIC_LIBRARIES += libskia_ext libjpeg-turbo-static libsimd \
 	libippj libippi libipps libippcore
-LOCAL_SHARED_LIBRARIES += libva libva-android libhardware libcutils libmix_imagedecoder libutils
+LOCAL_SHARED_LIBRARIES += libcutils libutils
 LOCAL_CFLAGS += -DADD_INTEL_SKIMAGEDECODER
 endif
 
diff --git a/include/core/SkImageDecoder.h b/include/core/SkImageDecoder.h
index f8f5a3f..30d6117 100644
--- a/include/core/SkImageDecoder.h
+++ b/include/core/SkImageDecoder.h
@@ -540,7 +540,6 @@ DECLARE_DECODER_CREATOR(GIFImageDecoder);
 DECLARE_DECODER_CREATOR(ICOImageDecoder);
 DECLARE_DECODER_CREATOR(JPEGImageDecoder);
 #ifdef ADD_INTEL_SKIMAGEDECODER
-DECLARE_DECODER_CREATOR(JPEGMixImageDecoder);
 DECLARE_DECODER_CREATOR(JPEGTurboImageDecoder);
 #endif
 DECLARE_DECODER_CREATOR(PNGImageDecoder);
diff --git a/src/images/SkForceLinking.cpp b/src/images/SkForceLinking.cpp
index 825241b..7239b9a 100644
--- a/src/images/SkForceLinking.cpp
+++ b/src/images/SkForceLinking.cpp
@@ -17,7 +17,6 @@ int SkForceLinking(bool doNotPassTrue) {
         CreateJPEGImageDecoder();
 #ifdef ADD_INTEL_SKIMAGEDECODER
         CreateJPEGTurboImageDecoder();
-        CreateJPEGMixImageDecoder();
 #endif
         CreateWEBPImageDecoder();
         CreateBMPImageDecoder();
-- 
1.7.9.5

